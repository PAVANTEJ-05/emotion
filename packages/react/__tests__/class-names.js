import 'test-utils/setup-env'
import React from 'react'
import { ClassNames, ThemeProvider } from '@emotion/react'
import { render } from '@testing-library/react'

test('css', () => {
  const { container } = render(
    <ClassNames>
      {({ css }) => (
        <div
          className={css`
            color: hotpink;
          `}
        />
      )}
    </ClassNames>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('should get the theme', () => {
  const { container } = render(
    <ThemeProvider theme={{ color: 'green' }}>
      <ClassNames>
        {({ css, theme }) => (
          <div
            className={css`
              color: ${theme.color};
            `}
          />
        )}
      </ClassNames>
    </ThemeProvider>
  )
  expect(container.firstChild).toMatchSnapshot()
})

test('cx', () => {
  const { container } = render(
    <ClassNames>
      {({ css, cx }) => {
        let secondClassButItsInsertedFirst = css`
          color: green;
        `
        let firstClassButItsInsertedSecond = css`
          color: hotpink;
        `

        return (
          <div
            className={cx(
              firstClassButItsInsertedSecond,
              'some-other-class',
              secondClassButItsInsertedFirst
            )}
          />
        )
      }}
    </ClassNames>
  )

  expect(container.firstChild).toMatchSnapshot()
})

test('css and cx throws when used after render', () => {
  let cx, css
  render(
    <ClassNames>
      {arg => {
        ;({ cx, css } = arg)
        return null
      }}
    </ClassNames>
  )

  expect(cx).toThrowErrorMatchingInlineSnapshot(
    `"cx can only be used during render"`
  )
  expect(css).toThrowErrorMatchingInlineSnapshot(
    `"css can only be used during render"`
  )
})
