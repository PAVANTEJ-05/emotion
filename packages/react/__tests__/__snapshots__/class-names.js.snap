// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`css 1`] = `
.emotion-0 {
  color: hotpink;
}

<div
  class="emotion-0"
/>
`;

exports[`cx 1`] = `
.emotion-0 {
  color: hotpink;
  color: green;
}

<div
  class="some-other-class emotion-0"
/>
`;

exports[`should get the theme 1`] = `
.emotion-0 {
  color: green;
}

<div
  class="emotion-0"
/>
`;
